File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ParsingInputs.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class ParsingInputs {
    fun parseInput(sdk: BindingLiquidSdk) {
        // ANCHOR: parse-inputs
        val input = "an input to be parsed..."

        try {
            val inputType = sdk.parse(input)
            when (inputType) {
                is InputType.BitcoinAddress -> {
                    println("Input is Bitcoin address ${inputType.address.address}")
                }
                is InputType.Bolt11 -> {
                    val amountStr = inputType.invoice.amountMsat?.toString() ?: "unknown"
                    println("Input is BOLT11 invoice for $amountStr msats")
                }
                is InputType.LnUrlPay -> {
                    println("Input is LNURL-Pay/Lightning address accepting min/max " +
                           "${inputType.data.minSendable}/${inputType.data.maxSendable} msats - BIP353 was used: ${inputType.bip353Address != null}")
                }
                is InputType.LnUrlWithdraw -> {
                    println("Input is LNURL-Withdraw for min/max " +
                           "${inputType.data.minWithdrawable}/${inputType.data.maxWithdrawable} msats")
                }
                else -> {
                    // Handle other input types
                } 
            }
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: parse-inputs
    }

    fun configureParsers() {
        // ANCHOR: configure-external-parser
        val mnemonic = "<mnemonic words>"

        // Create the default config, providing your Breez API key
        val config : Config = defaultConfig(LiquidNetwork.MAINNET, "<your Breez API key>")

        // Configure external parsers
        config.externalInputParsers = listOf(
            ExternalInputParser(
                providerId = "provider_a",
                inputRegex = "^provider_a",
                parserUrl = "https://parser-domain.com/parser?input=<input>"
            ),
            ExternalInputParser(
                providerId = "provider_b", 
                inputRegex = "^provider_b",
                parserUrl = "https://parser-domain.com/parser?input=<input>"
            )
        )

        try {
            val connectRequest = ConnectRequest(config, mnemonic)
            val sdk = connect(connectRequest)
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: configure-external-parser
    }
}

File: snippets/wasm/parsing_inputs.ts
import { defaultConfig, connect, type BindingLiquidSdk } from '@breeztech/breez-sdk-liquid'

const parseInputs = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: parse-inputs
  const input = 'an input to be parsed...'

  const parsed = await sdk.parse(input)

  switch (parsed.type) {
    case 'bitcoinAddress':
      console.log(`Input is Bitcoin address ${parsed.address.address}`)
      break

    case 'bolt11':
      console.log(
        `Input is BOLT11 invoice for ${
          parsed.invoice.amountMsat != null ? parsed.invoice.amountMsat.toString() : 'unknown'
        } msats`
      )
      break

    case 'lnUrlPay':
      console.log(
        `Input is LNURL-Pay/Lightning address accepting min/max ${parsed.data.minSendable}/${
          parsed.data.maxSendable
        } msats - BIP353 was used: ${parsed.bip353Address != null}`
      )
      break

    case 'lnUrlWithdraw':
      console.log(
        `Input is LNURL-Withdraw for min/max ${parsed.data.minWithdrawable}/${parsed.data.maxWithdrawable} msats`
      )
      break

    default:
      // Other input types are available
      break
  }
  // ANCHOR_END: parse-inputs
}

const configureParsers = async () => {
  // ANCHOR: configure-external-parser
  const mnemonic = '<mnemonics words>'

  // Create the default config, providing your Breez API key
  const config = defaultConfig('mainnet', '<your-Breez-API-key>')

  // Configure external parsers
  config.externalInputParsers = [
    {
      providerId: 'provider_a',
      inputRegex: '^provider_a',
      parserUrl: 'https://parser-domain.com/parser?input=<input>'
    },
    {
      providerId: 'provider_b',
      inputRegex: '^provider_b',
      parserUrl: 'https://parser-domain.com/parser?input=<input>'
    }
  ]

  await connect({ mnemonic, config })
  // ANCHOR_END: configure-external-parser
}
